pkg/distro/packagesets/loader.go

```diff
@@ -34,10 +38,12 @@ func Load(it distro.ImageType, replacements map[string]string) rpmmd.PackageSet
 	archName := arch.Name()
 	distribution := arch.Distro()
 	distroNameVer := distribution.Name()
-	distroName := strings.SplitN(distroNameVer, "-", 2)[0]
+	// we need to split from the right for "centos-stream-10" like
+	// distro names, sadly go has no rsplit() so we do it manually
+	distroName := distroNameVer[:strings.LastIndex(distroNameVer, "-")]
```

Comment:
I think that the best approach would be to use [`distroidparser.Parser`](https://github.com/osbuild/images/blob/9eb2318dfadc1060751a59872734202d33bb81b9/pkg/distroidparser/idparser.go#L34-L51) instead of this code.  Or even better, extend the `distro.Distribution` interface to return `distro.ID`. Anyway, it's probably material for a follow-up.

Reply:
Sadly distroidparser cannot be imported because of import cycles (I would like that fixed but that requires a bit more refactor). I like `distro.ID` in the distro.Distribution interface but for that I think we should remove Name/ReleaseVer/OsVersion along the way as its not orthogonal to ID() (and we need to make sure distro.ID covers all the cases thtat the other three cover (I think it does)). But that is probably a bigger discussion :)